distributed/rpc/batch/parameter_server.py

```diff
@@ -0,0 +1,144 @@
+import os
+import threading
+from datetime import datetime
+
+import torch
+import torch.distributed.rpc as rpc
+import torch.multiprocessing as mp
+import torch.nn as nn
+from torch import optim
+
+import torchvision
+
+
+batch_size = 20
+image_w = 64
+image_h = 64
+num_classes = 30
+batch_update_size = 5
+
+
+def timed_log(text):
+    print(f"{datetime.now().strftime('%H:%M:%S')} {text}")
+
+
+class BatchUpdateParameterServer(object):
+
+    def __init__(self, batch_update_size=batch_update_size):
+        self.model = torchvision.models.resnet50(num_classes=num_classes)
+        self.lock = threading.Lock()
+        self.future_model = torch.futures.Future()
+        self.batch_update_size = batch_update_size
+        self.curr_update_size = 0
+        self.optimizer = optim.SGD(self.model.parameters(), lr=0.001, momentum=0.9)
+        for p in self.model.parameters():
+            p.grad = torch.zeros_like(p)
+
+    def get_model(self):
+        return self.model
+
+    @staticmethod
+    @rpc.functions.async_execution
+    def update_and_fetch_model(ps_rref, grads):
+        self = ps_rref.local_value()
+        with self.lock:
+            self.curr_update_size += 1
+            timed_log(f"PS got one update, {self.curr_update_size} in batch")
```

Comment:
Nit: Add `batch_update_size` for the logging information so users have an idea of how many more updates it's blocked on?